### PR TITLE
Fix Bug 1453479 - more dark theme updates

### PR DESCRIPTION
**Pull request by rlr**

### CODE REVIEW

**Comment by rlr** [comment]
I have a few outstanding questions to Amy still. But maybe getting more eyeballs on this will create more? @Mardak 

screen shot 2018-04-11 at 11 33 26 pm
screen shot 2018-04-12 at 8 14 57 am
screen shot 2018-04-12 at 8 20 47 am
screen shot 2018-04-12 at 8 18 54 am
screen shot 2018-04-12 at 8 17 21 am

File: system-addon/content-src/components/Search/_Search.scss [hunk 1]
```diff
@@ -13,15 +13,12 @@
   cursor: default;
   display: flex;
   height: $search-height;
-  // The extra 1px is to account for the box-shadow being outside of the element
-  // instead of inside. It needs to be like that to not overlap the inner background
-  // color of the hover state of the submit button.
```
Comment (rlr) [inline system-addon/content-src/components/Search/_Search.scss]: I tried to reproduce this code comment but I couldnt see a difference? So I removed the horizontal margin so the dropdown is the right width.

**Comment by rlr** [comment]
@Mardak ok, I think this is ready for 👁 👁 . Hopefully the last "major" round of changes for now 😝  #tedious

File: system-addon/content-src/styles/_theme.scss [hunk 2]
```diff
@@ -16,10 +17,13 @@ body {
   --newtab-text-conditional-color: $grey-60;
   --newtab-text-primary-color: $grey-90;
   --newtab-text-secondary-color: $grey-50;
+  --newtab-textbox-background-color: $white;
   --newtab-textbox-border: $grey-90-20;
-  --newtab-textbox-color: inherit;
+  --newtab-textbox-focus-color: $blue-60;
+  --newtab-textbox-focus-boxshadow: 0 0 0 1px $blue-60, 0 0 0 4px rgba($blue-60, 0.3); // sass-lint:disable-line no-color-literals
```
Comment (rlr) [inline system-addon/content-src/styles/_theme.scss]: I don't like this but... is there a cleaner solution?

Comment (Mardak) [reply system-addon/content-src/styles/_theme.scss]: Are you referring to the duplicate stuff between default and dark? Maybe something like:
```diff
diff --git a/system-addon/content-src/styles/_theme.scss b/system-addon/content-src/styles/_theme.scss
index eb04f699..97bccdcc 100644
--- a/system-addon/content-src/styles/_theme.scss
+++ b/system-addon/content-src/styles/_theme.scss
@@ -1 +1,6 @@
+@mixin textbox-focus($color) {
+  --newtab-textbox-focus-color: $color;
+  --newtab-textbox-focus-boxshadow: 0 0 0 1px $color, 0 0 0 4px rgba($color, 0.3); // sass-lint:disable-line no-color-literals
+}
+
 // Default theme
@@ -21,4 +26,3 @@ body {
   --newtab-textbox-border: $grey-90-20;
-  --newtab-textbox-focus-color: $blue-60;
-  --newtab-textbox-focus-boxshadow: 0 0 0 1px $blue-60, 0 0 0 4px rgba($blue-60, 0.3); // sass-lint:disable-line no-color-literals
+  @include textbox-focus($blue-60);
 
@@ -74,4 +78,3 @@ body {
   --newtab-textbox-border: $grey-10-20;
-  --newtab-textbox-focus-color: $blue-40;
-  --newtab-textbox-focus-boxshadow: 0 0 0 1px $blue-40, 0 0 0 4px rgba($blue-40, 0.3); // sass-lint:disable-line no-color-literals
+  @include textbox-focus($blue-40);
 
```

File: system-addon/content-src/components/CollapsibleSection/_CollapsibleSection.scss [hunk 3]
```diff
@@ -73,7 +73,7 @@
   }
 
   &.active {
-    background: var(--newtab-element-active-color);
+    background: var(--newtab-element-hover-color);
```
Comment (Mardak) [inline system-addon/content-src/components/CollapsibleSection/_CollapsibleSection.scss]: Somewhat confusing that `.active` uses `-hover-color`.. oh well

File: system-addon/content-src/components/Search/_Search.scss (hunk 1 above)
Comment (Mardak) [reply system-addon/content-src/components/Search/_Search.scss]: Hmm.. @k88hudson added that comment in #3785.. Doesn't look like either 1px need to be around anymore?

File: system-addon/content-src/styles/_theme.scss [hunk 4]
```diff
@@ -63,43 +70,49 @@ body {
...
-$shadow-secondary: 0 1px 4px 0 $grey-90-10;
+$shadow-secondary: 0 1px 4px 0 $grey-90-20;
+$shadow-card: var(--newtab-card-shadow);
 $input-border: solid 1px var(--newtab-textbox-border);
-$input-border-active: solid 1px $blue-50;
+$input-border-active: solid 1px var(--newtab-textbox-focus-color);
 $input-error-border: solid 1px $red-60;
 $input-error-boxshadow: 0 0 0 1px $red-60, 0 0 0 4px rgba($red-60, 0.3);
-$input-focus-boxshadow: 0 0 0 1px $blue-50, 0 0 0 4px rgba($blue-50, 0.3);
+$input-focus-boxshadow: var(--newtab-textbox-focus-boxshadow);
```
Comment (Mardak) [inline system-addon/content-src/styles/_theme.scss]: If it's just the `var`, probably just directly use `var(--newtab-textbox-focus-boxshadow)` and `var(--newtab-card-shadow)` without the sass variable. Functions wouldn't really be able to do stuff with the value anyway.

File: system-addon/content-src/styles/activity-stream-windows.scss [hunk 5]
```diff
@@ -3,6 +3,5 @@
 $os-infopanel-arrow-height: 10px;
 $os-infopanel-arrow-offset-end: 6px;
 $os-infopanel-arrow-width: 20px;
-$os-search-focus-shadow-radius: 1px;
```
Comment (Mardak) [inline system-addon/content-src/styles/activity-stream-windows.scss]: Is this removal actually desired? It regresses #3745 which said to have 1px for windows.

File: system-addon/content-src/styles/_theme.scss [hunk 6]
```diff
@@ -7,6 +7,7 @@ body {
   --newtab-button-primary-color: $blue-60;
   --newtab-button-secondary-color: inherit;
   --newtab-element-active-color: $grey-20;
+  --newtab-element-hover-color: $grey-20;
```
Comment (Mardak) [inline system-addon/content-src/styles/_theme.scss]: There should be no differentiation from hover and clicking on a context menu item for default theme?

File: system-addon/content-src/styles/activity-stream-windows.scss (hunk 5 above)
Comment (rlr) [reply system-addon/content-src/styles/activity-stream-windows.scss]: I asked Amy explicitly about it. I'll double confirm.

File: system-addon/content-src/styles/_theme.scss (hunk 6 above)
Comment (rlr) [reply system-addon/content-src/styles/_theme.scss]: Oh right, it needs to get darker. 👍

File: system-addon/content-src/styles/activity-stream-windows.scss (hunk 5 above)
Comment (rlr) [reply system-addon/content-src/styles/activity-stream-windows.scss]: the UX powers say we can get rid of the windows specific special case 🎉

File: system-addon/content-src/components/CollapsibleSection/_CollapsibleSection.scss (hunk 3 above)
Comment (rlr) [reply system-addon/content-src/components/CollapsibleSection/_CollapsibleSection.scss]: we can rename the css class to 🤔 something.

File: system-addon/content-src/styles/_theme.scss [hunk 7]
```diff
@@ -63,43 +74,47 @@ body {
...
+  --newtab-card-shadow: 0 1px 8px 0 $grey-90-20;
 }
 
 // scss variables related to the theme.
 $border-primary: 1px solid var(--newtab-border-primary-color);
 $border-secondary: 1px solid var(--newtab-border-secondary-color);
 $shadow-primary: 0 0 0 5px var(--newtab-card-active-outline-color);
-$shadow-secondary: 0 1px 4px 0 $grey-90-10;
+$shadow-secondary: 0 1px 4px 0 $grey-90-20;
+$shadow-card: var(--newtab-card-shadow);
```
Comment (Mardak) [inline system-addon/content-src/styles/_theme.scss]: This sass variable can go too

File: system-addon/content-src/components/Search/_Search.scss [hunk 8]
```diff
@@ -13,17 +11,13 @@
...
-  margin: 1px 1px $section-spacing;
+  margin: 1px 0 $section-spacing;
```
Comment (Mardak) [inline system-addon/content-src/components/Search/_Search.scss]: k88 says if it looks right, then good to go, and from my testing, it seems fine with `margin-bottom: $section-spacing;`

File: system-addon/content-src/components/CollapsibleSection/_CollapsibleSection.scss (hunk 3 above)
Comment (Mardak) [reply system-addon/content-src/components/CollapsibleSection/_CollapsibleSection.scss]: Eh, we can just leave it :p

**Comment by rlr** [comment]
^ I also added `:active` state for the prefs gear icon

File: system-addon/content-src/styles/_theme.scss [hunk 9]
```diff
@@ -63,43 +74,46 @@ body {
...
 $border-secondary: 1px solid var(--newtab-border-secondary-color);
 $shadow-primary: 0 0 0 5px var(--newtab-card-active-outline-color);
-$shadow-secondary: 0 1px 4px 0 $grey-90-10;
+$shadow-secondary: 0 1px 4px 0 $grey-90-20;
 $input-border: solid 1px var(--newtab-textbox-border);
-$input-border-active: solid 1px $blue-50;
+$input-border-active: solid 1px var(--newtab-textbox-focus-color);
 $input-error-border: solid 1px $red-60;
 $input-error-boxshadow: 0 0 0 1px $red-60, 0 0 0 4px rgba($red-60, 0.3);
-$input-focus-boxshadow: 0 0 0 1px $blue-50, 0 0 0 4px rgba($blue-50, 0.3);
```
Comment (Mardak) [inline system-addon/content-src/styles/_theme.scss]: Oh, I forgot to note that I think sass allows a custom function where we can pass in a $red-60 or $blue-40/60/$color to generate the `0 0 0 1px $color, 0 0 0 4px rgba($color, 0.3)` value.. maybe.......... ?

**Pull request merged**